lib/ansible/modules/database/postgresql/postgresql_tablespace.py

```diff
@@ -0,0 +1,557 @@
+'''
+
+RETURN = r'''
+queries:
+    description: List of queries that was tried to be executed.
+    returned: always
+    type: str
+    sample: [ "CREATE TABLESPACE bar LOCATION '/incredible/ssd'" ]
+tablespace:
+    description: Tablespace name.
+    returned: always
+    type: str
+    sample: 'ssd'
+owner:
+    description: Tablespace owner.
+    returned: always
+    type: str
+    sample: 'Bob'
+options:
+    description: Tablespace options.
+    returned: always
+    type: dict
+    sample: { 'random_page_cost': 1, 'seq_page_cost': 1 }
+location:
+    description: Path to the tablespace in the file system.
+    returned: always
+    type: str
+    sample: '/incredible/fast/ssd'
+newname:
+    description: New tablespace name
+    returned: if existent
+    type: str
+    sample: new_ssd
+state:
+    description: Tablespace state at the end of execution.
+    returned: always
+    type: str
+    sample: 'present'
+'''
+
+try:
+    import psycopg2
+    HAS_PSYCOPG2 = True
+except ImportError:
+    HAS_PSYCOPG2 = False
+
+import ansible.module_utils.postgres as pgutils
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible.module_utils.database import SQLParseError, pg_quote_identifier
+from ansible.module_utils.postgres import postgres_common_argument_spec
+from ansible.module_utils._text import to_native
+from ansible.module_utils.six import iteritems
+
+
+def connect_to_db(module, kw, autocommit=False):
+    try:
+        db_connection = psycopg2.connect(**kw)
+        if autocommit:
+            if psycopg2.__version__ >= '2.4.2':
+                db_connection.set_session(autocommit=True)
+            else:
+                db_connection.set_isolation_level(psycopg2.extensions.ISOLATION_LEVEL_AUTOCOMMIT)
+
+    except TypeError as e:
+        if 'sslrootcert' in e.args[0]:
+            module.fail_json(msg='Postgresql server must be at least '
+                                 'version 8.4 to support sslrootcert')
+
+        module.fail_json(msg="unable to connect to database: %s" % to_native(e))
+
+    except Exception as e:
+        module.fail_json(msg="unable to connect to database: %s" % to_native(e))
+
+    return db_connection
+
+
+class PgTablespace(object):
+    def __init__(self, module, cursor, name):
+        self.module = module
+        self.cursor = cursor
+        self.name = name
+        self.exists = False
+        self.owner = ''
+        self.settings = {}
+        self.location = ''
+        self.executed_queries = []
+        self.new_name = ''
+        self.opt_not_supported = False
+        # Collect info:
+        self.get_info()
+
+    def get_info(self):
+        # Check that spcoptions exists:
+        opt = self.__exec_sql("SELECT 1 FROM information_schema.columns "
+                              "WHERE table_name = 'pg_tablespace' "
+                              "AND column_name = 'spcoptions'", add_to_executed=False)
+
+        # For 9.1 version and earlier:
```

Comment:
PostgreSQL 9.3 and earlier versions are no longer maintained. Do we really need such conditions in new modules?

Reply:
I'm afraid CI tests will fail because they use even 9.0. Anyway it's not difficult to implement this, so I suggest to remain this condition.